addition of group > 1 in test and in backend

chachaleo opened the pull request:
### Description
I added the implementation for group > 1 in the back-end test implementation of conv transpose. And also added some tests with group > 1.

chachaleo committed:
addition of group > 1 in test and in backend

Signed-off-by: chachaleo <charlotte.leo@hotmail.com>

xadupre commented:
You should sign off your commits (git commit -s ...) to pass DCO test (click on the link details to sign your past commits). You also to run the following commands to update the documentation and the new test in the folder onnx/backend/test/data.

```bash
python onnx/backend/test/cmd_tools.py generate-data --clean
python  onnx/defs/gen_doc.py
python onnx/backend/test/stat_coverage.py 
```

chachaleo committed:
modif from review

Signed-off-by: chachaleo <charlotte.leo@hotmail.com>

xadupre commented:
Is it possible to restore the original onnx models whicj already exist and keep only the one you add? That way it is easier to see your changes.

chachaleo commented:
Sorry yes I will clean this. 

These two lines make an error : 
python  onnx/defs/gen_doc.py
python onnx/backend/test/stat_coverage.py 

even without my changes, do I have to work on another branch than main ?

xadupre commented:
What do you mean by `do I have to work on another branch than main`? Submitting a PR like you did is usually what people do.

justinchuby commented:
If I understand your question correctly, you meant if you need to create a diff against a branch other than main. Main is the current development branch. The issue maybe because a different version of protobuf / platform caused slightly different binary results for protobuf. You may include only the relavent changes in your commit by manually selecting them.

xadupre commented:
For any question related to git, some famous LLM have usually good answers to something like *how to revert to the previous commit in a subdirectory using Git*.

chachaleo commented:
Ok! :)

I actually can't manage to generate the node data, when I run 

`python onnx/backend/test/cmd_tools.py generate-data --clean`

It only generates the node data for the existing tests, and when I run 

`python onnx/backend/test/cmd_tools.py generate-data --clean -d`

it runs only the test of convtranspose but without the one I added.

 The only changes I made was the op_convtranspose implementation and I added my test in the convtranspose.py file in 
onnx>backend>test>case>node

What should I do to generate the data ?

xadupre commented:
I usually use this ``python onnx/backend/test/cmd_tools.py generate-data --clean`` to generate the data. Then I do ``git add ...`` with only the files I want to add to the PR. Then ``git commit -s -m ...``. Then I do ``git reset --hard`` to revert all the changes I do not want to include to the PR. Maybe some others have a more simple way to do it but that's what I do.

xadupre commented:
I created another PR with your changes https://github.com/onnx/onnx/pull/6175 and updated only the necessayr data, could you review it?
